kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java

```diff
@@ -76,24 +77,27 @@ public FileSystemViewKeyInputFormat(FileSystemDataset<E> dataset,
     Format format = dataset.getDescriptor().getFormat();
 
     boolean isSpecific = SpecificRecord.class.isAssignableFrom(dataset.getType());
+    String readerSchema = conf.get("kite.readerSchema");
+    if (isSpecific && readerSchema == null) {
```

Comment:
Should we be setting the reader schema if your data model is reflect or only for specific based on the type and for generic based on the reader schema?

Also, should we always be setting the reader schema? If the user doesn't set it, either by creating a view or by calling `withSchema` on the `DatasetKeyInputFormat`, we should probably set it to the schema of the dataset so that all records are read using the same schema, right?